### ISSUE TITLE
az rest command is returning incorrect values when querying via azure cli

### ISSUE THREAD

**Issue by garudapnx** [Azure/azure-cli#24244]
> ### `az feedback` auto-generates most of the information requested below, as of CLI version 2.0.62

**Related command**
az rest -m get -u /subscriptions/<subscriptionid>/providers/Microsoft.Web/sites?api-version=2022-03-01 2>&1 | grep -i '"id": "/subscriptions/<subscriptionid>/resourcegroups/<resourcegroupname>/' | wc -l

**Describe the bug**
When running the command it doesn't match the returned count of Web App in the specified subscription and resource group vs. what is in the Azure environment.

**To Reproduce**
N/A

**Expected behavior**
Expected behavior should match the returned count vs. the web app count that is in the Azure environment.

**Environment summary**
Azure CLI version 2.41.0

**Additional context**

<img width="1495" alt="azcli" src="https://user-images.githubusercontent.com/116078509/199529942-3716785d-bc36-4b05-9b43-e592de6eba99.png">

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by RakeshMohanMSFT** [comment]
@garudapnx I see in your command you are trying to get the word count, is that correct?

**Comment by garudapnx** [comment]
@RakeshMohanMSFT, yes correct just a quick matching of what is in the environment and what is the returned count from the command.

**Comment by RakeshMohanMSFT** [comment]
@garudapnx do you happened to have logic apps/function apps as well?
Because  "type": "Microsoft.Web/sites" can mean all of those in that particular resource group?

**Comment by garudapnx** [comment]
> @garudapnx do you happened to have logic apps/function apps as well? Because "type": "Microsoft.Web/sites" can mean all of those in that particular resource group?

There are no logic app in this subscription only App Services.

**Comment by garudapnx** [comment]
Would you be able to figure out what is wrong with this command? Because it obviously doesn't reflect the returned result vs. the environment.

**Comment by RakeshMohanMSFT** [comment]
@garudapnx Can you please let us know what is the count you are expecting and what is that you are getting? Also share your subscription details to my email ramaruth[at]microsoft.com

**Comment by garudapnx** [comment]
> @garudapnx Can you please let us know what is the count you are expecting and what is that you are getting? Also share your subscription details to my email ramaruth[at]microsoft.com

Hi Rakesh, it is in the screenshot that I uploaded. If you can see in the resource group there are 249 App Service but in the command returned it is only 22. Do you still need me to send it to your email?

**Comment by RakeshMohanMSFT** [comment]
@garudapnx Yes, i need your subscription id and the user account used to access these.

**Comment by garudapnx** [comment]
> @garudapnx Yes, i need your subscription id and the user account used to access these.

How can I message you privately? This is a public thread and publishing the subscription id and and account here is against our security policy.

**Comment by RakeshMohanMSFT** [comment]
> @garudapnx Can you please let us know what is the count you are expecting and what is that you are getting? Also share your subscription details to my email ramaruth[at]microsoft.com

Send it to my email

**Comment by garudapnx** [comment]
Ok then, maybe let's continue to talk in your email. I will send you an email.

**Comment by RakeshMohanMSFT** [comment]
@jiasli We need your help here, for some reason the rest command is returning wrong number of resources in the cloud shell.

**Comment by garudapnx** [comment]
Hi Guys, I just came back from long paternity leave. So do we have update on this issue? This has been sitting for quite sometime without true resolution.

**Comment by jiasli** [comment]
It is possibly because the response contains a [`nextLink`](https://learn.microsoft.com/en-us/rest/api/azure/#async-operations-throttling-and-paging) property. You may check the response to see if the `nextLink` property exists. One strange thing is the document for `nextLink` states:

> Typically, the response includes the nextLink property when the list operation returns more than 1,000 items.

But the number 249 is far below that 1000 threshold.
 
`az rest` is only for calling raw REST API and it doesn't support paging. If the result does contain `nextLink` property, you have to implement your own logic to fetch the URL in `nextLink`.

For simplicity, we recommend using `az webapp list` which handles paging for you.
